lib/reek/smells/smell_repository.rb

```diff
@@ -23,37 +25,32 @@ def initialize(smell_types: self.class.smell_types,
                      configuration: {})
         @configuration = configuration
         @smell_types   = smell_types
+        @detectors     = smell_types.map do |klass|
```

Comment:
I'm sorry to say this, but sometimes we are really overusing the memoization "||=" pattern in `Reek`.
This kind of initialization belongs in the ..... initializer, and not in a method IMHO.